### PR TITLE
Move normal distribution to Aten(CPU)

### LINKED ISSUE #24746: Migrate `normal` and `normal_` from the TH to Aten (CPU)
Porting TH operators is essential for code simplicity and performance reasons.

Porting guides and Q&A are available in umbrella issue: #24507

Feel free to add @VitalyFedyunin as a reviewer to get a prioritized review.

### PR DESCRIPTION
**Pull request by XiaobingSuper**
Fix https://github.com/pytorch/pytorch/issues/24746

### CODE REVIEW

**Comment by kostmo** [comment]
## :pill: CircleCI build failures summary and remediations
As of commit 227415f8:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=227415f8ab022ba3a54867331b1bb65fdeb15e79).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/4199829) pytorch_linux_xenial_py3_6_gcc5_4_test (1/1)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=83107415) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=83107415))
<details>
<summary>
<code>Jan 10 09:12:41 RuntimeError: test_autograd failed!</code>
</summary>

```
Jan 10 09:12:41 Ran 1032 tests in 217.441s 
Jan 10 09:12:41  
Jan 10 09:12:41 FAILED (errors=2, skipped=12, expected failures=1) 
Jan 10 09:12:41  
Jan 10 09:12:41 Generating XML reports... 
Jan 10 09:12:41 Traceback (most recent call last): 
Jan 10 09:12:41   File "test/run_test.py", line 456, in <module> 
Jan 10 09:12:41     main() 
Jan 10 09:12:41   File "test/run_test.py", line 449, in main 
Jan 10 09:12:41     raise RuntimeError(message) 
Jan 10 09:12:41 RuntimeError: test_autograd failed! 
Jan 10 09:12:41 + cleanup 
Jan 10 09:12:41 + retcode=1 
Jan 10 09:12:41 + set +x 
Jan 10 09:12:41 =================== sccache compilation log =================== 
Jan 10 09:12:41 =========== If your build fails, please take a look at the log above for possible reasons =========== 
Jan 10 09:12:41 Compile requests                  7 
Jan 10 09:12:41 Compile requests executed         6 
Jan 10 09:12:41 Cache hits                        0 
Jan 10 09:12:41 Cache misses                      6 
Jan 10 09:12:41 Cache timeouts                    0 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

File: aten/src/ATen/native/Distributions.cpp [hunk 1]
```diff
@@ -197,6 +197,118 @@ Tensor& geometric_(Tensor& self, double p, Generator* gen) {
...
+}
+
+Tensor normal_cpu(const Tensor& mean, double std, Generator* gen) {
+  Tensor ret = at::empty_like(mean, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
+  normal_out_cpu(ret, mean, std, gen);
+  return ret;
+}
+
+Tensor normal_cpu(double mean, const Tensor& std, Generator* gen) {
+  Tensor ret = at::empty_like(std, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
```
Comment (ezyang) [inline aten/src/ATen/native/Distributions.cpp]: @VitalyFedyunin When we do ports like this, should we really be continuing to use the `LEGACY_CONTIGUOUS_MEMORY_FORMAT` here?

File: aten/src/ATen/native/cpu/UnaryOpsKernel.cpp [hunk 2]
```diff
@@ -415,6 +469,7 @@ REGISTER_DISPATCH(sigmoid_stub, &sigmoid_kernel);
 REGISTER_DISPATCH(bernoulli_mkl_stub, &bernoulli_mkl_kernel);
 REGISTER_DISPATCH(cauchy_stub, &cauchy_kernel);
 REGISTER_DISPATCH(geometric_stub, &geometric_kernel);
+REGISTER_DISPATCH(normal_stub, &normal_kernel);
```
Comment (ezyang) [inline aten/src/ATen/native/cpu/UnaryOpsKernel.cpp]: For example, this code is handling AVX properly!

**Review by ezyang — CHANGES_REQUESTED** [review]
AVX code is in the wrong place

File: aten/src/ATen/native/Distributions.cpp (hunk 1 above)
Comment (VitalyFedyunin) [reply aten/src/ATen/native/Distributions.cpp]: Agree, this is unnecessary and we should omit this argument or use MemoryFormat::Contiguous when it is required to have contiguous tensor.

In this particular case `ret` is expected to be Contiguous so `MemoryFormat::Contiguous` is required.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**